localization/ja/decorator/README.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Decorator
+category: Structural
+language: ja
+tag:
+ - Gang of Four
+ - Extensibility
+---
+
+## 別名
+
+Wrapper
+
+## 目的
+
+オブジェクトに責任を動的に追加する。
+Decorator パターンは、サブクラス化よりも柔軟な機能拡張方法を提供する。
+
+## 説明
+
+実世界の例
+
+> 近くの丘に住む怒ったトロールがいる。大抵は素手だが、たまに武器を持っている。トロールに武器を持たせるためには、新しいトロールを作る必要はなく、適切な武器を動的に装飾すればいい。
+
+分かりやすい説明
+
+> Decorator パターンを用いると、decorator クラスのオブジェクトにラップすることで、オブジェクトの振る舞いを実行時間で動的に変更することができる。
+
+Wikipedia の説明
+
+> オブジェクト指向プログラミングにおいて、Decorator パターンは、同じクラスの別のオブジェクトの振る舞いに影響を与えることなく、静的もしくは動的に、個別のオブジェクトに振る舞いを加えることができるデザインパターンである。Decorator パターンは、単一責任の原則を遵守するためにしばしば便利である。なぜなら、クラスの機能性を変更せずに拡張することで、開放/閉鎖原則などの関心事の特異な分野を元に、クラスの間で機能性を分割することができるためである。
+
+**サンプルコード**
+
+トロールを例として使用する。まず、Troll のインターフェイスを実装する SimpleTroll クラスを作る。
+
+```java
+public interface Troll {
+  void attack();
+  int getAttackPower();
+  void fleeBattle();
+}
+
+@Slf4j
+public class SimpleTroll implements Troll {
+
+  @Override
+  public void attack() {
+    LOGGER.info("The troll tries to grab you!");
+  }
+
+  @Override
+  public int getAttackPower() {
+    return 10;
+  }
+
+  @Override
+  public void fleeBattle() {
+    LOGGER.info("The troll shrieks in horror and runs away!");
+  }
+}
+```
+
+次に、トロールに対してこん棒を追加する。decorator を使い、動的に追加することが可能である。
+
+```java
+@Slf4j
+public class ClubbedTroll implements Troll {
+
+  private final Troll decorated;
+
+  public ClubbedTroll(Troll decorated) {
+    this.decorated = decorated;
+  }
+
+  @Override
+  public void attack() {
+    decorated.attack();
+    LOGGER.info("The troll swings at you with a club!");
+  }
+
+  @Override
+  public int getAttackPower() {
+    return decorated.getAttackPower() + 10;
+  }
+
+  @Override
+  public void fleeBattle() {
+    decorated.fleeBattle();
+  }
+}
+```
+
+以下がトロールの動き。
+
+```java
+// simple troll
+LOGGER.info("A simple looking troll approaches.");
+var troll = new SimpleTroll();
+troll.attack();
+troll.fleeBattle();
+LOGGER.info("Simple troll power: {}.\n", troll.getAttackPower());
+
+// change the behavior of the simple troll by adding a decorator
+LOGGER.info("A troll with huge club surprises you.");
+var clubbedTroll = new ClubbedTroll(troll);
+clubbedTroll.attack();
+clubbedTroll.fleeBattle();
+LOGGER.info("Clubbed troll power: {}.\n", clubbedTroll.getAttackPower());
+```
+
+アウトプット
+
+```java
+A simple looking troll approaches.
+The troll tries to grab you!
+The troll shrieks in horror and runs away!
+Simple troll power: 10.
+
+A troll with huge club surprises you.
+The troll tries to grab you!
+The troll swings at you with a club!
+The troll shrieks in horror and runs away!
+Clubbed troll power: 20.
+```
+
+## クラス図
+
+![alt text](./etc/decorator.urm.png "Decorator pattern class diagram")
```

Comment:
Could you check that the image is found in this location? I think it currently resides in `localization/ja/decorator/decorator.urm.png`

Reply:
@iluwatar 
Oops... Sorry about that. I moved the image under `etc` directory

https://github.com/iluwatar/java-design-patterns/pull/2444/commits/2832ffa4461868153178a7969f036f689ce56cc1